core/templates/dev/head/pages/community-dashboard-page/community-dashboard-page.controller.ts

```diff
@@ -0,0 +1,105 @@
+
+/**
+ * @fileoverview Directive for the community dashboard page.
+ */
+
+require('base_components/BaseContentDirective.ts');
+require(
+  'components/common-layout-directives/common-elements/' +
+  'background-banner.directive.ts');
+require(
+  'pages/community-dashboard-page/translation-opportunities/' +
+  'translation-opportunities.directive.ts');
+require(
+  'pages/community-dashboard-page/voiceover-opportunities/' +
+  'voiceover-opportunities.directive.ts');
+
+require('domain/utilities/LanguageUtilService.ts');
+require('domain/utilities/UrlInterpolationService.ts');
+
+require(
+  'pages/community-dashboard-page/community-dashboard-page.constants.ajs.ts');
+
+angular.module('oppia').directive('communityDashboardPage', [
+  'UrlInterpolationService', function(
+      UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/pages/community-dashboard-page/' +
+      'community-dashboard-page.directive.html'),
+      controllerAs: '$ctrl',
+      controller: [
+        '$window', 'LanguageUtilService', 'TranslationLanguageService',
+        'COMMUNITY_DASHBOARD_TABS_DETAILS',
+        'DEFAULT_OPPORTUNITY_LANGUAGE_CODE',
+        function(
+            $window, LanguageUtilService, TranslationLanguageService,
+            COMMUNITY_DASHBOARD_TABS_DETAILS,
+            DEFAULT_OPPORTUNITY_LANGUAGE_CODE) {
+          var ctrl = this;
+          var LAST_SELECTED_TRANSLATION_LANGUAGE = (
+            'last_selected_translation_lang');
+          var prevLanguageCode = $window.localStorage.getItem(
+            LAST_SELECTED_TRANSLATION_LANGUAGE);
+          var allAudioLanguageCodes = LanguageUtilService
+            .getAllVoiceoverLanguageCodes();
+
+          ctrl.languageCodesAndDescriptions = (
+            allAudioLanguageCodes.map(function(languageCode) {
+              return {
+                id: languageCode,
+                description: (
+                  LanguageUtilService.getAudioLanguageDescription(
+                    languageCode))
+              };
+            }));
+          ctrl.languageCode = (
+          allAudioLanguageCodes.indexOf(prevLanguageCode) !== -1 ?
+            prevLanguageCode : DEFAULT_OPPORTUNITY_LANGUAGE_CODE);
+
+          TranslationLanguageService.setActiveLanguageCode(ctrl.languageCode);
+
+          ctrl.onChangeLanguage = function() {
+            TranslationLanguageService.setActiveLanguageCode(ctrl.languageCode);
+            $window.localStorage.setItem(
+              LAST_SELECTED_TRANSLATION_LANGUAGE, ctrl.languageCode);
+          };
+
+          ctrl.showLanguageSelector = function() {
+            if (!ctrl.tabsDetails[ctrl.activeTabName].hasOwnProperty(
+              'customizationOptions')) {
+              return false;
+            }
+            return (
+              ctrl.tabsDetails[ctrl.activeTabName].customizationOptions.indexOf(
```

Comment:
It might not be a bad idea to declare a tabDetails variable at the outset of this function and use that, to prevent awkwardly long clauses like this one.